src/Compilers/CSharp/Portable/Binder/DecisionDagBuilder_ListPatterns.cs

```diff
@@ -53,7 +55,14 @@ private Tests MakeTestsAndBindingsForListPattern(BoundDagTemp input, BoundListPa
 
                         if (slice.Pattern is BoundPattern slicePattern)
                         {
-                            var sliceEvaluation = new BoundDagSliceEvaluation(slicePattern.Syntax, slicePattern.InputType, lengthTemp, startIndex: startIndex, endIndex: index, slice.IndexerAccess, slice.SliceMethod, input);
+                            Debug.Assert(slice.IndexerAccess is not null);
+                            Debug.Assert(index <= 0);
+                            Debug.Assert(slice.ReceiverPlaceholder is not null);
+                            Debug.Assert(slice.ArgumentPlaceholder is not null);
+
+                            var sliceEvaluation = new BoundDagSliceEvaluation(slicePattern.Syntax, slicePattern.InputType, lengthTemp, startIndex: startIndex, endIndex: index,
+                                slice.IndexerAccess, slice.ReceiverPlaceholder, slice.ArgumentPlaceholder, input);
```

Comment:
>slice.IndexerAccess [](http://example.com/codeflow?start=32&length=19)

Consider asserting what kind of node this can be since the comment for BoundDagSliceEvaluation narrows the set of nodes that should be expected by comparison to BoundSlicePattern